assets/src/dashboard/app/views/templates/detail.js

```diff
@@ -93,6 +99,28 @@ function TemplateDetail() {
     };
   }, [template]);
 
+  const activeTemplateIndex = useMemo(
+    () => templates?.findIndex((t) => t.id === template?.id),
+    [template, templates]
+  );
+
+  const handleNextTemplateClick = useCallback(() => {
+    const nextIndex = clamp(activeTemplateIndex + 1, [0, templates.length - 1]);
+    setTemplate(templates[nextIndex]);
+  }, [activeTemplateIndex, templates]);
+
+  const handlePrevTemplateClick = useCallback(() => {
```

Comment:
Since these two click handlers are almost identical, perhaps extract into a function that's used by both? To DRY things up a little.

Reply:
BTW, should navigation using the arrows update the route too? It currently stays at `#/template-detail?id=1` when navigating through the templates.

Reply:
Yes!  Good callouts, I'll dry up the code and update the history!!